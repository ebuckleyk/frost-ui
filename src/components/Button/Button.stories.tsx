import * as React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { Loader2, Mail } from 'lucide-react';

import { Button } from './Button';

function WithIconDemo() {
  return (
    <Button>
      <Mail className="mr-2 h-4 w-4" /> Login with Email
    </Button>
  );
}

function LoadingDemo() {
  return (
    <Button disabled>
      <Loader2 className="mr-2 h-4 w-4 animate-spin" />
      Please wait
    </Button>
  );
}

type ComponentType = React.ComponentProps<typeof Button> & { text?: string };
const meta: Meta<ComponentType> = {
  component: Button,
  render: ({ text: buttonText, ...args }) => <Button {...args}>{buttonText}</Button>,
  argTypes: {
    variant: {
      control: { type: 'radio' },
    },
  },
};

export default meta;

type Story = StoryObj<ComponentType>;

export const Demo: Story = {
  args: {
    text: 'Button',
  },
};

export const WithIcon: Story = {
  render: WithIconDemo,
};

export const WithLoader: Story = {
  render: LoadingDemo,
};
