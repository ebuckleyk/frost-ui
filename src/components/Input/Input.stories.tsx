import * as React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { Input } from './Input';
import { Label } from '../Label';
import { Card } from '../Card';

type ComponentType = typeof Input;
const meta: Meta<ComponentType> = {
  component: Input,
  render: () => (
    <Card className="flex items-center p-5">
      <Input />
    </Card>
  ),
};

export default meta;

type Story = StoryObj<ComponentType>;

export const Demo: Story = {
  args: {
    placeholder: 'Placeholder text',
  },
};

export const SelectWithLabel: Story = {
  render: (args) => {
    return (
      <Card className="p-10">
        <Label htmlFor="input">Input</Label>
        <Input {...args} id="input" placeholder="Placeholder text" />
      </Card>
    );
  },
};
