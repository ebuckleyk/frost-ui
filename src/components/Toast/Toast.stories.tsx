import * as React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import { Toast } from './Toast';

type ComponentType = React.ComponentProps<typeof Toast>;
const meta: Meta<ComponentType> = {
  component: Toast,
};

export default meta;

type Story = StoryObj<ComponentType>;
export const Default: Story = {};
