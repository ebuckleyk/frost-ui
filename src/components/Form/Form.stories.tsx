import * as React from 'react';
import { zodResolver } from '@hookform/resolvers/zod';
import { Meta, StoryObj } from '@storybook/react';
import { useForm } from 'react-hook-form';
import * as z from 'zod';

import { Button } from '../Button';
import { Input } from '../Input';
import { toast } from '../Toast';
import { Form, FormControl, FormDescription, FormField, FormItem, FormLabel, FormMessage } from './Form';

const formSchema = z.object({
  username: z.string().min(2).max(25),
});

function ProfileFormDemo() {
  // 1. Define your form.
  const form = useForm<z.infer<typeof formSchema>>({
    resolver: zodResolver(formSchema),
    defaultValues: {
      username: '',
    },
  });
  // 2. Define a submit handler
  function onSubmit(values: z.infer<typeof formSchema>) {
    toast({
      title: 'You submitted',
      description: (
        <pre className="mt-2 w-[340px] rounded-md bg-slate-950 p-4">
          <code>{JSON.stringify(values, null, 2)}</code>
        </pre>
      ),
    });
  }
  return (
    <Form {...form}>
      <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-8">
        <FormField
          control={form.control}
          name="username"
          render={({ field }) => (
            <FormItem>
              <FormLabel>Username</FormLabel>
              <FormControl>
                <Input placeholder="ebuckleyk" {...field} />
              </FormControl>
              <FormDescription>This is your public display name.</FormDescription>
              <FormMessage />
            </FormItem>
          )}
        />
        <Button type="submit">Submit</Button>
      </form>
    </Form>
  );
}

type ComponentType = React.ComponentProps<typeof Form>;
const meta: Meta<ComponentType> = {
  component: Form,
  render: ProfileFormDemo,
};

export default meta;

type Story = StoryObj<ComponentType>;
export const Demo: Story = {};
