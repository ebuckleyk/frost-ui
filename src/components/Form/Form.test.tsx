import * as React from 'react';
import { zodResolver } from '@hookform/resolvers/zod';
import { render } from '@testing-library/react';
import { useForm } from 'react-hook-form';
import * as z from 'zod';

import { Button } from '../Button';
import { Input } from '../Input';
import { toast } from '../Toast';
import { Form, FormControl, FormDescription, FormField, FormItem, FormLabel, FormMessage } from './Form';

const formSchema = z.object({
  username: z.string().min(2).max(25),
});

function Component() {
  // 1. Define your form.
  const form = useForm<z.infer<typeof formSchema>>({
    resolver: zodResolver(formSchema),
    defaultValues: {
      username: '',
    },
  });
  // 2. Define a submit handler
  function onSubmit(values: z.infer<typeof formSchema>) {
    toast({
      title: 'You submitted',
      description: (
        <pre className="mt-2 w-[340px] rounded-md bg-slate-950 p-4">
          <code>{JSON.stringify(values, null, 2)}</code>
        </pre>
      ),
    });
  }
  return (
    <Form {...form}>
      <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-8">
        <FormField
          control={form.control}
          name="username"
          render={({ field }) => (
            <FormItem>
              <FormLabel>Username</FormLabel>
              <FormControl>
                <Input placeholder="ebuckleyk" {...field} />
              </FormControl>
              <FormDescription>This is your public display name.</FormDescription>
              <FormMessage />
            </FormItem>
          )}
        />
        <Button type="submit">Submit</Button>
      </form>
    </Form>
  );
}

describe('Form', () => {
  it('should render Form component to match snapshot', () => {
    const result = render(<Component />);
    expect(result).toMatchSnapshot();
  });
});
