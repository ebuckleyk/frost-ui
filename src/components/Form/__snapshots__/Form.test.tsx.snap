// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form > should render Form component to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="space-y-8"
      >
        <div
          class="space-y-2"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for=":r0:-form-item"
          >
            Username
          </label>
          <input
            aria-describedby=":r0:-form-item-description"
            aria-invalid="false"
            class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
            id=":r0:-form-item"
            name="username"
            placeholder="ebuckleyk"
            value=""
          />
          <p
            class="text-[0.8rem] text-muted-foreground"
            id=":r0:-form-item-description"
          >
            This is your public display name.
          </p>
        </div>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background backdrop-blur-md backdrop-saturate-100 transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary/80 text-primary-foreground hover:bg-primary/60 h-10 px-4 py-2"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="space-y-8"
    >
      <div
        class="space-y-2"
      >
        <label
          class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
          for=":r0:-form-item"
        >
          Username
        </label>
        <input
          aria-describedby=":r0:-form-item-description"
          aria-invalid="false"
          class="flex h-9 w-full rounded-md border border-input bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
          id=":r0:-form-item"
          name="username"
          placeholder="ebuckleyk"
          value=""
        />
        <p
          class="text-[0.8rem] text-muted-foreground"
          id=":r0:-form-item-description"
        >
          This is your public display name.
        </p>
      </div>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background backdrop-blur-md backdrop-saturate-100 transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary/80 text-primary-foreground hover:bg-primary/60 h-10 px-4 py-2"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
